core/clustersmngr/cluster/single.go

```diff
@@ -75,6 +75,8 @@ func getImpersonatedConfig(config *rest.Config, user *auth.UserPrincipal) (*rest
 		return nil, fmt.Errorf("no user ID or Token found in UserPrincipal")
 	} else if tok := user.Token(); tok != "" {
 		cfg.BearerToken = tok
+		// Clear the token file as it seems to take precedence over the token
+		cfg.BearerTokenFile = ""
```

Comment:
It feels like there's an opportunity for some refactoring to remove the duplication?

https://github.com/weaveworks/weave-gitops/blob/8d316717a22809bca21fe1ebc0b4671c27ff75e2/pkg/kube/config_getter.go#L40

Reply:
Good catch!